recipe/gen_patch_json.py

```diff
@@ -1830,10 +1830,28 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
 
         # older versions of dask-cuda do not work on non-UNIX operating systems and must be constrained to UNIX
         # issues in click 8.1.0 cause failures for older versions of dask-cuda
-        if record_name == "dask-cuda" and record.get("timestamp", 0) <= 1645130882435:  # 22.2.0 and prior
-            new_depends = record.get("depends", [])
-            new_depends += ["click ==8.0.4", "__linux"]
-            record["depends"] = new_depends
+        if record_name == "dask-cuda":
+            # older versions of dask-cuda do not work on non-UNIX operating systems and must be constrained to UNIX
+            # issues in click 8.1.0 cause failures for older versions of dask-cuda
+            if record.get("timestamp", 0) <= 1645130882435:  # 22.2.0 and prior
+                new_depends = record.get("depends", [])
+                new_depends += ["click ==8.0.4", "__linux"]
+                record["depends"] = new_depends
+
+            # there are various inconsistencies between the pinnings of dask-cuda on `rapidsai` and `conda-forge`,
+            # this makes the packages roughly consistent while also removing the python upper bound where present
+            if record["version"] == "0.18.0":
+                _replace_pin("dask >=2.9.0", "dask >=2.4.0", record["depends"], record)
+            elif record["version"] == "0.19.0":
+                _replace_pin("dask >=2.9.0", "dask >=2.22.0,<=2021.4.0", record["depends"], record)
+                _replace_pin("distributed >=2.18.0", "distributed >=2.22.0,<=2021.4.0", record["depends"], record)
+            elif record["version"] == "21.6.0":
+                _replace_pin("distributed >=2.22.0,<=2021.5.1", "distributed >=2.22.0,<2021.5.1", record["depends"], record)
+            elif record["version"] in ("21.10.0", "22.2.0"):
```

Comment:
Seems we are missing 21.8.0. Should that have a version constraint as well?

Reply:
Looking through the diff, 21.8.0 didn't have any pinning differences from the RAPIDS-published package.